code-style/action.yml

```diff
@@ -130,12 +195,14 @@ runs:
     - name: "Install pre-commit"
       shell: bash
       run: |
+        source .venv/bin/activate
         python -m pip install pre-commit
         pre-commit install
 
     - name: "Run pre-commit"
       shell: bash
       run: |
+        source .venv/bin/activate
```

Comment:
Do we need to activate the venv every time? Just curious if this is a consequence when using Poetry.

Reply:
We have to do it because we don't want to pollute / use the system installation. Otherwise, we might end up installing in the wrong place which leads to the issues in #559.
Since we work with a virtual environment, you have to activate it at each step you require it because each step is like a fresh shell. This means that not doing it in a step `B` leads to working with the system instead of your virtual environment, even if you activated it in the previous step `A`.

Reply:
Alternatively, you can run the python binary directly in `./venv/bin/python` or `./venv/scripts/python` in Windows. I find this a bit awkward though, and you can end up with very long lines if you're calling pytest or sphinx with a whole load of arguments, for example.